exploration/number-selection.md

```diff
@@ -0,0 +1,200 @@
+It also customized this selection beyond the capabilities of `com.ibm.icu.text.PluralRules`
+by allowing for explicit value matching and an `offset` parameter.
+
+As pointed out by <a href="https://github.com/mihnita">@mihnita</a> in particular,
+category selection is not always appropriate for selection on a number:
+the number may be representing some completely other quantity,
+such as a four-digit year or the integer value of an enumerator.
+
+Furthermore, as pointed out by <a href="https://github.com/ryzokuken">@ryzokuken</a>
+in <a href="https://github.com/unicode-org/message-format-wg/pull/457#discussion_r1307443288">#457 (comment)</a>,
+ordinal selection works similarly to plural selection, but uses a separate set of rules
+for each locale.
+This is visible in English, for example, where plural rules use only `one` and `other`
+but ordinal rules use `one` (_1st_, _21st_, etc.), `few` (_2nd_, _22nd_, etc.), 
+`many` (_3rd_, _23rd_, etc.), and `other` (_4th_, _5th_, etc.).
+
+Additionally,
+MF1 provides `ChoiceFormat` selection based on a complex rule set
+(and which allows determining if a number falls into a specific range).
+
+Both JS and ICU PluralRules implementations provide for determining the plural category
+of a range based on its start and end values.
+Range-based selectors are not initially considered here.
+
+## Use-Cases
+
+As a user, I want to write messages that use the correct plural for
+my locale and enables translation to locales that use different rules.
+
+As a user, I want to write messages that use the correct ordinal for
+my locale and enables translation to locales that use different rules.
+
+As a user, I want to write messages in which the pattern used depends on exactly matching
+a numeric value.
+
+As a user, I want to write messages that mix exact matching and 
+either plural or ordinal selection in a single message. 
+> For example:
+>```
+>.match {$numRemaining}
+>0 {{You have no more chances remaining (exact match)}}
+>1 {{You have one more chance remaining (exact match)}}
+>one {{You have {$numRemaining} chance remaining (plural)}}
+> * {{You have {$numRemaining} chances remaining (plural)}}
+>```
+
+
+## Requirements
+
+- Enable cardinal plural selection.
+- Enable ordinal number selection.
+- Enable exact match selection.
+- Enable simultaneous "exact match" and either plural or ordinal selection.
+  > For example, allow variants `[1]` and `[one]` in the same message.
+- Selection must support relevant formatting options, such as `minimumFractionDigits`.
+  > For example, in English the value `1` matches plural rule `one` while the value `1.0`
+  > matches the plural rule `other` (`*`)
+- Encourage developers to provide the formatting options used in patterns to the selector
+  so that proper selection can be done.
+
+## Constraints
+
+ICU MF1 messages using `plural` and `selectordinal` should be representable in MF2.
+
+## Proposed Design
+
+Given that we already have a `:number`,
```

Comment:
> Given that we already have a `:number`,
> it makes sense to add a `<matchSignature>` to it with an option


Sorry, but it is not that obvious.
`:number` is not a question, so one can't decide on it.

You have to ask a question about the number.

"Hey, `:number 43` , true or false?
If you answer correctly you win 1000$"
That means nothing. It is not a good question.

`is :number 43 even`" is a question, you can answer.
`is :number 43 prime`" is a question, you can answer.
`is :number 43 plural`" is a question, you can answer.

Reply:
There are two questions here.

The first is: should the `:number` function used for formatting also do selection? @eemeli has already pointed out that we want users to use the same configuration for selection (e.g. with `:plural`) as will ultimately be used to format the number. This message produces grammatically inferior results:

```
.local $tenths = {$num :number minFractionDigits=1}
.match {$num :plural}
one {{You have {$tenths} value}}  // won't be seen
* {{You have {$tenths} values}}
```

The second question is: if `:number` is a selector, what sort of selector is it by default? @eemeli answers that `:plural` is the most common usage. Alternatives would be `:ordinal` and whatever we call exact match. I agree with @eemeli that `:plural` is the far-and-away most common case and thus should be the default.

So... @mihnita are you objecting to the phrasing here or the conclusion? If the phrasing, do you want to suggest alternate text?

Reply:
@aphillips Your "won't be seen" comment is a bit misleading, as the problem with that message is that the `one` variant _may_ indeed be seen, formatted as "You have 1.0 value".

Reply:
You're right.. that comment goes against what I'm trying to say, so I fixed it.

Reply:
Since we already defined `$tenths` and use it in the body of the options, then I expect `:plural` on it:
```
.local $tenths = {$num :number minFractionDigits=1}
.match {$tenths :plural}
one {{You have {$tenths} value}}  // not desirable
* {{You have {$tenths} values}}
```

And not doing that is easy to do at lint time.